### DISCUSSION TITLE
Postgres entity generation support base tables.

### DISCUSSION THREAD

**Discussion by DenuxPlays** [SeaQL/sea-orm#2412] [Ideas]
Hey,

when I execute this sql file to generate my schema:
```sql

CREATE TABLE base
(
    id BIGINT PRIMARY KEY
);

CREATE TABLE full_schema
(
    another_field TEXT,
    PRIMARY KEY (id)
) INHERITS (base);
```

and I then generate the entities only the base schema entity is being generated.
I guess that full support for base schemas is a bit too much and too specific for an orm but for entity generation every table that uses the `INHERITS` statement should be generated as an entity and the base table should be ignored.

In this example a only one struct like this should be generated:
```rust
struct FullSchema {
   id:  i64,
   another_field: String
}
```

Edit:

This feature is on Postgres [Don't do this](https://wiki.postgresql.org/wiki/Don%27t_Do_This) list so it would be pointless to add support for a legacy feature. So

**Comment by alfalcon90** [comment]
+1 Would love this as well. I submitted a [feature request](https://github.com/SeaQL/sea-orm/issues/2743) before I saw this post.

**Comment by yuriy-yarosh** [comment]
PostgreSQL is so old, that it had accumulated a lot of legacy and poorly supported features, over time.

There's an explicit [Don't do this](https://wiki.postgresql.org/wiki/Don%27t_Do_This) list, and Inheritance is one of the common pitfalls.

Sure, you can use it, but that does not really mean you should be using it.

**Reply by DenuxPlays** [reply]
Yes I learned this too and totally forgot to update my discussion.

Sorry for that

**Reply by yuriy-yarosh** [reply]
Nah, it's fine.
